doc/MSTP/MSTP.md

```diff
@@ -0,0 +1,1178 @@
+  - [MSTP global disable](#mstp-global-disable)
+  - [MSTP region name/version change](#mstp-region-nameversion-change)
+  - [Instance creation](#instance-creation)
+  - [Instance deletion](#instance-deletion)
+  - [MSTP Instance creation](#mstp-Instance-Creation)
+  - [MSTP Instance deletion](#mstp-Instance-Deletion)
+  - [Add VLAN to Exisiting Instance](#add-vlan-existing)
+  - [Delete VLAN to Exisiting Instance](#delete-vlan-existing)
+  - [Add VLAN member](#add-vlan-member)
+  - [Del VLAN member](#del-vlan-member)
+* [Configuration Commands](#configuration-commands)
+  - [Global Level](#global-level)
+  - [Region Level](#region-level)
+  - [Instance, Interface Level](#instance-interface-level)
+  - [Show Commands](#show-commands)
+  - [Clear Commands](#clear-commands)
+  - [Debug Commands](#debug-commands)
+  - [Disabled Commands](#disabled-commands)
+* [YANG Model](#yang-model)
+* [Rest API Support](#rest-api-support)
+* [Warm Boot](#warm-boot)
+* [Testing Requirements](#testing-requirements)
+  - [Unit test cases](#unit-test-cases)
+* [References](#references)
+
+# Scope
+This document describes the High Level Design of Multiple Spanning Tree Protocol.
+
+# Abbreviations
+
+|**Term**|**Meaning**|
+| :- | :- |
+|BPDU|Bridge Protocol Data Unit|
+|CIST|Common Internal Spanning Tree|
+|CST|Common Spanning Tree|
+|IST|Internal Spanning Tree|
+|MSTI|Multiple Spanning Tree Instance|
+|MSTID|Multiple Spanning Tree Identifier|
+|MSTP|Multiple Spanning Tree Protocol|
+|STP|Spanning Tree Protocol|
+|VLAN|Virtual Local Area Network|
+|VID|VLAN identifier|
+
+# Overview
+Multiple Spanning Tree Protocol (MSTP) enhances the Spanning Tree Protocol (STP) by enabling the creation of multiple spanning tree instances within a network. It provides a mechanism to map VLANs to specific spanning tree instances which offers network segmentation and improved control over traffic flow.
+
+# Introduction
+
+Spanning Tree Protocol (STP) is a network protocol that operates at the Data Link layer (Layer 2) of the OSI model. Its primary purpose is to prevent loops in a Local Area Network (LAN). Network loops can cause broadcast storms and multiple frame copies, leading to network inefficiencies and failures. STP ensures a loop-free topology by creating a spanning tree that selectively blocks some network paths. 
+
+Multiple Spanning Tree Protocol (MSTP), standardized in IEEE 802.1s, is an extension of STP. MSTP enhances STP by providing better loop prevention, path redundancy, and optimal bandwidth utilization. MSTP allows network engineers to create multiple spanning trees, each associated with a set of VLANs, and assign them to different switch ports. This configuration optimizes bandwidth usage, reduces convergence time, and simplifies network management. 
+
+ 
+MSTP reduces convergence time compared to STP. When a network topology change occurs, only the affected MSTI needs to reconverge, minimizing the impact on the entire network. 
+
+
+
```

Comment:
Mention kernel level changes expected to enable MSTP, if no changes expected and following PVST behavior, mention it accordingly.

Reply:
We will update the HLD.